docs/tutorials/sparse/csr.md

```diff
@@ -0,0 +1,338 @@
+
+## Compressed Sparse Row Matrix
+
+A CSRNDArray represents a 2D matrix as three separate 1D arrays: **data**, **indptr** and **indices**, where the column indices for row `i` are stored in `indices[indptr[i]:indptr[i+1]]` in ascending order, and their corresponding values are stored in `data[indptr[i]:indptr[i+1]]`.
+
+- **data**: CSR format data array of the matrix
+- **indices**: CSR format index array of the matrix
+- **indptr**: CSR format index pointer array of the matrix
+
+### Example Matrix Compression
+
+For example, given the matrix:
+```
+[[7, 0, 8, 0]
+ [0, 0, 0, 0]
+ [0, 9, 0, 0]]
+```
+
+We can compress this matrix using CSR, and to do so we need to calculate `data`, `indices`, and `indptr`.
+
+The `data` array holds all the non-zero entries of the matrix in row-major order. Put another way, you create a data array that has all of the zeros removed from the matrix, row by row, storing the numbers in that order. Your result:
+
+    data = [7, 8, 9]
+
+The `indices` array stores the column index for each non-zero element in `data`. As you cycle through the data array, starting with 7, you can see it is in column 0. Then looking at 8, you can see it is in column 2. Lastly 9 is in column 1. Your result:
+
+    indices = [0, 2, 1]
+
+The `indptr` array is what will help identify the rows where the data appears. It stores the index into `data` of the first non-zero element number of each row of the matrix. This array always starts with 0 (reasons can be explored later), so indptr[0] is 0. Each subsequent value in the array is the aggregate number of non-zero elements up to that row. Looking at the first row of the matrix you can see two non-zero values, so indptr[1] is 2. The next row contains all zeros, so the aggregate is still 2, so indptr[2] is 2. Finally, you see the last row contains one non-zero element bring the aggregate to 3, so indptr[3] is 3. Your result:
+
+    indptr = [0, 2, 2, 3]
+
+Note that in MXNet, the column indices for a given row are always sorted in ascending order,
+and duplicated column indices for the same row are not allowed.
+
+## Array Creation
+
+There are a few different ways to create a `CSRNDArray`, but first let's recreate the matrix we just discussed using the `data`, `indices`, and `indptr` we calculated.
+
+* We can create a CSRNDArray with data, indices and indptr by using the `csr_matrix` function:
+
+
+```python
+import mxnet as mx
+# Create a CSRNDArray with python lists
+shape = (3, 4)
+data_list = [7, 8, 9]
+indptr_list = [0, 2, 2, 3]
+indices_list = [0, 2, 1]
+a = mx.nd.sparse.csr_matrix(data_list, indptr_list, indices_list, shape)
+# Inspect the matrix
+a.asnumpy()
+```
+
+
+```python
+import numpy as np
+# Create a CSRNDArray with numpy arrays
+data_np = np.array([7, 8, 9])
+indptr_np = np.array([0, 2, 2, 3])
+indices_np = np.array([0, 2, 1])
+b = mx.nd.sparse.csr_matrix(data_np, indptr_np, indices_np, shape)
+b.asnumpy()
```

Comment:
We were debating including the outputs. Since the current convention had them removed we stuck with that, but I think there's value in including the outputs for when the tutorial is converted to html and the reader may not be in the position to generate the results. Having the output allows the reader to follow along regardless of their access to a working instance of mxnet.